test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,672 @@
+            else if (range_ret == WT_NOTFOUND && normal_ret == 0) {
+                testutil_check(normal_cursor->get_key(normal_cursor.get(), &normal_key));
+                std::string normal_key_str = normal_key;
+                /*  Make sure that normal cursor returns a key that is outside of the range. */
+                if (next) {
+                    testutil_assert(upper_bound.key.length() != 0);
+                    testutil_assert(custom_lexicographical_compare(
+                                      normal_key_str, upper_bound.key, true) == false);
+                } else {
+                    testutil_assert(lower_bound.key.length() != 0);
+                    testutil_assert(custom_lexicographical_compare(
+                                      normal_key_str, lower_bound.key, false) == true);
+                }
+                break;
+            }
+
+            std::string range_key_str = range_key;
+            if (next && upper_bound.key.length() != 0)
+                testutil_assert(custom_lexicographical_compare(
+                                  range_key_str, upper_bound.key, upper_bound.inclusive) == true);
+            else if (!next && lower_bound.key.length() != 0)
+                testutil_assert(custom_lexicographical_compare(
+                                  lower_bound.key, range_key_str, lower_bound.inclusive) == true);
+            /* Make sure that records match between both cursors. */
+            testutil_check(normal_cursor->get_key(normal_cursor.get(), &normal_key));
+            testutil_check(range_cursor->get_key(range_cursor.get(), &range_key));
+            testutil_assert(strcmp(range_key, normal_key) == 0);
+        }
+    }
+
+    /*
+     * Use the random generator either set no bounds, only lower bounds, only upper bounds or both
+     * bounds on the range cursor. The lower and upper bounds are randomly generated strings as
+     * well. The inclusive configuration is also randomly set as well.
+     */
+    std::pair<bound, bound>
+    set_random_bounds(test_harness::thread_context *tc, scoped_cursor &range_cursor)
+    {
+        bool set_lower_inclusive, set_upper_inclusive;
+        int64_t key_size;
+        std::string lower_key, upper_key;
+
+        auto set_random_bounds = random_generator::instance().generate_integer(0, 3);
+        if (set_random_bounds == NO_BOUNDS)
+            range_cursor->bound(range_cursor.get(), "action=clear");
+
+        if (set_random_bounds == LOWER_BOUND_SET || set_random_bounds == ALL_BOUNDS_SET) {
+            set_lower_inclusive = random_generator::instance().generate_integer(0, 1);
+            key_size =
+              random_generator::instance().generate_integer(static_cast<int64_t>(1), tc->key_size);
+            lower_key = random_generator::instance().generate_random_string(
+              key_size, characters_type::ALPHABET);
+            // Reverse case
+            lower_key = std::string(tc->key_size, 'z');
```

Comment:
I realised that we still need to hardcode this because range cursor would still return keys outside of the set bounds, because range cursor is not implemented yet.